lib/features/mailbox/domain/extensions/list_mailbox_extension.dart

```diff
@@ -23,4 +26,21 @@ extension ListMailboxExtension on List<PresentationMailbox> {
     //   return 0;
     // });
   }
+}
+
+extension ListMailboxExtensions on List<Mailbox> {
+  Tuple2<List<PresentationMailbox>, List<PresentationMailbox>> splitMailboxList(bool test(Mailbox element)) {
+    final validBuilder = ListBuilder<PresentationMailbox>();
+    final invalidBuilder = ListBuilder<PresentationMailbox>();
+    forEach((element) {
+      if (test(element)) {
+        validBuilder.add(element.toPresentationMailbox());
+      } else {
+        invalidBuilder.add(element.toPresentationMailbox());
+      }
+    });
+    return Tuple2(
+        validBuilder.build().toList(),
+        validBuilder.build().toList());
```

Comment:
IMO, should be `invalidBuilder` here. Please add unit-test for this function also